### ISSUE TITLE
database.yml failing to load figaro.env

### ISSUE THREAD

**Issue by ender4** [laserlemon/figaro#202]
I am running rails 4.1.8 with figaro 1.1.1 on Windows. When using the rails console trying to access the postgres database results in an error (PG::ConnectionBad: fe_sendauth: no password supplied).

config/application.yml looks like this:
development:
  postgres_username: '******'
  postgres_password: '******'

and config/database.yml looks like this:
development:
  ...
  postgres_username: <%= Figaro.env.postgres_username %>
  postgres_password: <%= Figaro.env.postgres_password %>

Do I need to use a different verision of either rails or figaro? Is this a recurrence [issue #70](https://github.com/laserlemon/figaro/issues/70)?

**Comment by ender4** [comment]
Problem was on my end.

**Comment by howdoicomputer** [comment]
@ender4 Which was...?

**Comment by tannerj** [comment]
I'm not sure what @ender4 did to solve his problem, but I've had a couple of issues getting this to work properly. Some of them are rather strange, I'm not sure of a better place to put this info, so I'll put it here. The first problem I had, which @ender4 addresses above, is that you must put your ENV['VARIABLE'] inside <%= %> erb brackets. The next issue I had is the main reason I'm writting this comment. **If, like me, you decide to start your database password with an '!', things get really fun.** if you only use one set of quotes and try to embed from config/application.yml to config/database.yml

in application.yml
`
DB_PASS: "!asdfadsf"
`

in database.yml
`
password: <%= ENV['DB_PASS'] %>
`

The quotes you put around DB_PASS in config/application.yml get stripped away and your left with

`
password: !ajdkfasdfdf
`

This is not what we want because starting a yaml variable with an '!' creates a yaml tag, http://www.yaml.org/spec/1.2/spec.html#id2761292, and that will not get properly feed into your database.

In order to get around this issue, we have to nest double quotes inside single quotes to ensure there are still quotes around the value when it is parsed from database.yml.

config/application.yml
`
password: '"!adsfadsfadsf
`

 I hope this saves someone time in the future.

TL;DR
If your password begins with an '!', you must double quote it.
`
DB_PASS: '"!asdklfadjslfk"'
`
